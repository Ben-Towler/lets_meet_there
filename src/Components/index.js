import SiteHeader from './SiteHeader/SiteHeader.component';
import SearchForm from './SearchForm/SearchForm.component';
import Trip from './Trip/Trip.component';
import Button from './Button/Button.component';
import Form from './Form/Form.component';
import Flight from './Flight/Flight.component';
import FlightSummary from './FlightSummary/FlightSummary.component';
import BookingLink from './BookingLink/BookingLink.component';
import Spinner from './Spinner/Spinner.component';
import LocationSearch from './LocationSearch/locationSearch.component.tsx';

export {
  SiteHeader,
  BookingLink,
  SearchForm,
  Trip,
  Flight,
  FlightSummary,
  Button,
  Form,
  Spinner,
  LocationSearch
}