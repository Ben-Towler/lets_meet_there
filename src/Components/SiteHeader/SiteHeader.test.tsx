import React from 'react';
import { SiteHeader } from 'Components';
import { BrowserRouter as Router } from 'react-router-dom';
import { configure } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import { render } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import {userMocks} from 'Mocks/data.mock';
configure({adapter: new Adapter()});

describe('Site Header', () => {

  test('snapshot matches', () => {
    const renderer = ShallowRenderer.createRenderer()
    const result = renderer.render(
      <SiteHeader 
        user={null} 
      />
    )
    expect(result).toMatchSnapshot()
  });

  test('renders home link', () => {
    const { getByText } = render(
      <Router>
        <SiteHeader />
      </Router>
    );
    const link = getByText('Home');
    expect(link).toBeInTheDocument();
  });

  test('renders profile and signout when logged in', () => {
    const { getByText } = render(
      <Router>
        <SiteHeader user={userMocks} />
      </Router>
    );
    const link = getByText('Profile');
    expect(link).toBeInTheDocument();
    const signout = getByText('Sign Out');
    expect(signout).toBeInTheDocument();
  });

  test('should render and fire a sign out button component', () => {
    const mockFunction = jest.fn();
    const {getByText} = render( <Router>
                                  <SiteHeader user={userMocks} />
                                </Router>)
    const btn = getByText('Sign Out');
    btn.onclick = mockFunction
    btn.click()
    expect(mockFunction).toHaveBeenCalled();
    expect(btn).toBeInTheDocument();
  })

  test('should render and fire a sign in button component', () => {
    const mockFunction = jest.fn();
    const {getByText} = render( <Router>
                                  <SiteHeader />
                                </Router>)

    const btn = getByText('Sign In with Google');
    btn.onclick = mockFunction
    btn.click()
    expect(mockFunction).toHaveBeenCalled();
    expect(btn).toBeInTheDocument();
  })
});

