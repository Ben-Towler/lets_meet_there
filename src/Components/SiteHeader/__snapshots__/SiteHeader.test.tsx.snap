// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Header snapshot matches 1`] = `
<ForwardRef(styled.header)>
  <ForwardRef
    to="/"
  >
    Home
  </ForwardRef>
  <Button
    id="signin"
    onClick={[Function]}
  >
    Sign In with Google
  </Button>
</ForwardRef(styled.header)>
`;
